central_v2/src/components/GameCard.tsx

```diff
@@ -0,0 +1,270 @@
+    height: buttonCount > 3 ? '48px' : '24px', // Height if more than 3 buttons on medium and larger screens
+  },
+}));
+
+const ButtonWrapper = styled(Box)(({ theme }) => ({
+  display: 'flex',
+  gap: '4px',
+  position: 'absolute',
+  bottom: 0,
+  left: 0,
+  flexWrap: 'wrap',
+
+  [theme.breakpoints.down('sm')]: {
+    justifyContent: 'start',
+    '> *:nth-of-type(n + 4)': {
+      marginTop: '4px',
+      marginLeft: '0',
+    },
+  },
+
+  [theme.breakpoints.up('md')]: {
+    justifyContent: 'start',
+    '> *:nth-of-type(n + 5)': {
+      marginTop: '4px',
+      marginLeft: '0',
+    },
+  },
+}));
+
+interface DescriptionTextProps {
+  buttonCount: number;
+}
+
+const DescriptionText = styled(Typography, {
+  shouldForwardProp: (prop) => prop !== 'buttonCount',
+})<DescriptionTextProps>(({ theme, buttonCount }) => ({
+  flex: 1,
+  height: 'auto',
+  fontFamily: 'Rubik',
+  fontWeight: '400',
+  fontSize: '16px',
+  lineHeight: '18.96px',
+  color: '#384466',
+  display: '-webkit-box',
+  overflow: 'hidden',
+  textOverflow: 'ellipsis',
+  WebkitBoxOrient: 'vertical',
+  WebkitLineClamp: buttonCount > 3 ? 4 : 5, 
+  [theme.breakpoints.up('sm')]: {
+    WebkitLineClamp: buttonCount > 2 ? 4 : 5, // 4 lines if > 3 buttons on medium or larger screens
+  },
+  [theme.breakpoints.up('md')]: {
+    WebkitLineClamp: buttonCount > 3 ? 4 : 5, // 4 lines if > 4 buttons on large screens
+  },
+}));
+
+const BottomButtonBox = styled(Box)(() => ({
+  width: '100%',
+  height: '38px',
+  gap: '8px',
+  display: 'flex',
+  flexDirection: 'row',
+  justifyContent: 'center',
+}));
+
+const PrimaryButton1 = styled(Button)(() => ({
+  width: 'auto',
+  height: '38px',
+  padding: '4px 12px',
+  gap: '8px',
+  borderRadius: '54px',
+  background: 'linear-gradient(270.1deg, #1C94C3 0.09%, #2A6AC6 64.33%, #2C62C6 76.27%, #3153C7 99.91%)',
+  boxShadow: '0px 5px 22px 0px rgba(71, 217, 255, 0.3)',
+  color: '#FFFFFF',
+  textTransform: 'none',
+}));
+
+const SecondaryButton = styled(Button)(() => ({
+  width: 'auto',
+  height: '24px',
+  padding: '4px 8px',
+  gap: '4px',
+  borderRadius: '12px',
+  background: 'linear-gradient(90deg, #E81144 0%, #E31C5E 100%)',
+  color: '#FFFFFF',
+  textTransform: 'none',
+  fontFamily: 'Rubik',
+  fontSize: '14px',
+  fontWeight: 400,
+  lineHeight: '16.59px',
+  textAlign: 'center',
+  boxShadow: '0px 3px 12px 0px #95002366',
+  zIndex: 2, 
+  boxSizing: 'border-box',
+  minWidth: '20px',
+}));
+
+function getDomainAndGrades(game: IGameTemplate) {
+  const extractedQuestions = game?.questionTemplates?.map((question) => question.questionTemplate);
+  const CCSSArray = extractedQuestions?.map((question) => {
+    return `${question.grade}.${question.domain}`;
+  });
+  return Array.from(new Set(CCSSArray));
+}
+
+export default function StyledGameCard({ id, title, description, image, apiClients, game }: StyledGameCardProps) {
+  const domainAndGrades = getDomainAndGrades(game);
+
+  return (
```

Comment:
This is so clean. I love the use of styled-components here to get our Render function super readable. Nice job